pandas/tests/io/parser/test_python_parser_only.py

```diff
@@ -314,3 +315,26 @@ def test_malformed_skipfooter(python_parser_only):
     msg = "Expected 3 fields in line 4, saw 5"
     with pytest.raises(ParserError, match=msg):
         parser.read_csv(StringIO(data), header=1, comment="#", skipfooter=1)
+
+
+def test_comment_char_in_default_value(python_parser_only):
```

Comment:
Can we run this test on the C engine too?

Reply:
Does not work with C engine currently as written above. Fix there is way more complex than here

Reply:
I saw that, but I think it would be good to `xfail` on the C engine so we're aware to keep working on a fix.

Reply:
Ah misunderstood you then. Will add the test

Reply:
added the c tests